### DISCUSSION TITLE
How to get latest form errors from zod validations in mode: 'all'

### DISCUSSION THREAD

**Discussion by JohannesKlauss** [react-hook-form/react-hook-form#8553] [Q&A]
I have this codesandbox:

[Codesandbox](https://codesandbox.io/s/old-dust-str80k?file=/src/App.tsx)

I am validating with zod that an array of objects holding numbers has a nonnegative required value.
And I am validating on that array that the sum of those numbers is 100.

I am listening to changes to the `formState.errors` to check if the latest errors occur there.

But when I for example enter the values 20, 20, 50 no errors appear, even though the sum is 90. Zod validation is triggered but the error object is stale. It also doesn't trigger errors when I enter the first value with the other two inputs empty (which should trigger an error for not filling required fields).

If I click on the submit button everything is correctly validated and populated, but not during onChange.

Is this a bug or am I missing something here?

**Comment by bluebill1049** [comment]
`formState` update will always at the render phase, not within onChange callback (next render)

**Reply by JohannesKlauss** [reply]
Is it somehow possible to get the current errors?

**Reply by bluebill1049** [reply]
give `getFieldState` a try?

**Reply by JohannesKlauss** [reply]
Hmh...., this doesn't force a re-render when the error state changes. What I need is a forced re-render when onChange triggers the validation and the validator returns a new result.

**Comment by floHoppen** [comment]
Hello,

I have a same problem, do you have an answer ?

thanks

**Reply by kevinashworth** [reply]
With zod and mode all, eventually I resorted to several of these (seemingly unnecessary) changes:
```
-              {...register("foo")}
+              {...register("foo", { onBlur: () => trigger(), onChange: () => trigger() })}
```
